crates/solidity/testing/snapshots/cst_output/EventDefinition/transfer/generated/0.4.11-success.yml

```diff
@@ -0,0 +1,37 @@
+# This file is generated automatically by infrastructure scripts. Please don't edit by hand.
+
+Source: >
+  1  │ event Transfer(address indexed from, address indexed to, uint256 value);         │ 0..72
```

Comment:
Question about error recovery, unrelated to this PR:

In the before (deleted) snapshot, shouldn't the `SeparetedBy` error recovery have recovered from the error on `from` here, to skip it, and recover after the following comma, to continue parsing the next parameter `address indexed to` correctly? Or is this NIY?

Reply:
If I recall correctly/not mixing things up, we couldn't properly recover from `NoMatch` elements in a `SeparatedBy` parser without risking misparses, unless we did the FIRST/FOLLOW computation that we decided to punt on for now. I had a counter-example but I don't have it handy right now...

Similar behaviour can be noticed here: https://github.com/Xanewok/slang/blob/8fe0b94218db07163f33e09cc4d15288e7f48b42/crates/solidity/testing/snapshots/cst_output/ContractMembers/separated_recovery/generated/0.4.11-failure.yml#L86

However, partial `IncompleteMatch` are recovered from until the separator, see https://github.com/Xanewok/slang/blob/8fe0b94218db07163f33e09cc4d15288e7f48b42/crates/solidity/testing/snapshots/cst_output/ContractMembers/separated_recovery/generated/0.4.11-failure.yml#L78

Then again, I would imagine a sequence of (TypeName, IndexedKeyword, Identifier) should be marked as "incomplete" (not empty/no match) if missing the suffix, so I may be misremembering the issue there. Definitely worth looking into it!

Reply:
Thanks for explaining!